### DISCUSSION TITLE
Dynamic API route returns 404

### DISCUSSION THREAD

**Discussion by PraveenPeri** [vercel/next.js#28341] [Help]
### What version of Next.js are you using?

10.1.3

### What version of Node.js are you using?

14.15.4

### What browser are you using?

Chrome/Edge

### What operating system are you using?

Linux

### How are you deploying your application?

next dev and next start

### Describe the Bug

Added a dynamic route under api folder like this api/entitySelection/[id].ts but the page hit to /api/entitySelection/1 always returns 404

### Expected Behavior

The path should return the content defined in the API at the route.

### To Reproduce

1. Add a dynamic route under a named folder inside api folder like /api/test/[id].ts
2. Define the API to return something like res.status(200).json({name: "John Doe"})
3. Run next dev and hit the URL http://localhost:3000/api/test/1 and see that it returns 404
4. Now add an index.ts api under test folder and see that hitting http://localhost:3000/api/test returns the expected content

**Comment by dikwickley** [comment]
Can you send the snippet of your api route?
Also #4 is the expected behaviour. (https://nextjs.org/docs/api-routes/dynamic-api-routes)

**Reply by PraveenPeri** [reply]
![image](https://user-images.githubusercontent.com/6571000/130289008-42272965-11ad-464c-bc83-3b987ee57e32.png)
**Code in [id].ts**
`import type { NextApiRequest, NextApiResponse } from 'next'`
`export default function OneSelection(req: NextApiRequest, res: NextApiResponse): void {`
   &nbsp;&nbsp;`//Return something for testing`
&nbsp;&nbsp;&nbsp;&nbsp;`res.status(200).json({`
       &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; `data: {`
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`id: "001", hrn: "hrn:hrs:001", name: "The Shawshank Redemption", year: 1994,`
               &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; `director: {`
                    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`name: "Frank Darabont",`
                    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`DOB: "01/28/1959"`
               &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; `}`
       &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;` }`
 &nbsp;&nbsp;&nbsp;&nbsp;  ` })`
`}`

**Behavior**
![image](https://user-images.githubusercontent.com/6571000/130289690-7018fdc2-a85c-41bc-831e-ad755c6ac8cf.png)

**Reply by PraveenPeri** [reply]
Wondering if this is related to [#26546](https://github.com/vercel/next.js/issues/26546)

**Reply by PraveenPeri** [reply]
@DiKwickley @timneutkens This issue's been moved to discussion but I believe this is a bug. If not, I would appreciate suggestion on how to resolve this.
